ksql-common/src/test/java/io/confluent/ksql/schema/ksql/LogicalSchemaTest.java

```diff
@@ -1047,6 +796,120 @@ public void shouldNotMatchRandomFieldNameAsBeingMetaOrKeyFields() {
     assertThat(SOME_SCHEMA.isKeyField("well_this_ain't_in_the_schema"), is(false));
   }
 
+  @Test
+  public void shouldThrowOnDuplicateKeyFieldName() {
+    // Given:
+    final Builder builder = LogicalSchema.builder()
+        .keyField("fieldName", SqlTypes.BIGINT);
+
+    // Then:
+    expectedException.expect(IllegalArgumentException.class);
+
+    // When:
+    builder.keyField("fieldName", SqlTypes.BIGINT);
+  }
+
+  @Test
+  public void shouldThrowOnDuplicateValueFieldName() {
+    // Given:
+    final Builder builder = LogicalSchema.builder()
+        .valueField("fieldName", SqlTypes.BIGINT);
+
+    // Then:
+    expectedException.expect(IllegalArgumentException.class);
+
+    // When:
+    builder.valueField("fieldName", SqlTypes.BIGINT);
+  }
+
+  @Test
+  public void shouldAllowKeyFieldsWithSameNameButDifferentSource() {
+    // Given:
+    final Builder builder = LogicalSchema.builder();
+
+    // When:
+    builder
+        .keyField("fieldName", SqlTypes.BIGINT)
+        .keyField(Field.of("source", "fieldName", SqlTypes.BIGINT))
+        .keyField(Field.of("diff", "fieldName", SqlTypes.BIGINT));
+
+    // Then:
+    assertThat(builder.build().keyFields(), contains(
+        Field.of("fieldName", SqlTypes.BIGINT),
+        Field.of("source", "fieldName", SqlTypes.BIGINT),
+        Field.of("diff", "fieldName", SqlTypes.BIGINT)
+    ));
+  }
+
+  @Test
+  public void shouldAllowValueFieldsWithSameNameButDifferentSource() {
+    // Given:
+    final Builder builder = LogicalSchema.builder();
+
+    // When:
+    builder
+        .valueField("fieldName", SqlTypes.BIGINT)
+        .valueField(Field.of("source", "fieldName", SqlTypes.BIGINT))
+        .valueField(Field.of("diff", "fieldName", SqlTypes.BIGINT));
+
+    // Then:
+    assertThat(builder.build().valueFields(), contains(
+        Field.of("fieldName", SqlTypes.BIGINT),
+        Field.of("source", "fieldName", SqlTypes.BIGINT),
+        Field.of("diff", "fieldName", SqlTypes.BIGINT)
+    ));
+  }
+
+  @Test
+  public void shouldGetKeySchema() {
+    // Given:
+    final LogicalSchema schema = LogicalSchema.builder()
+        .keyField("fieldName", SqlTypes.DOUBLE)
+        .keyField(Field.of("source", "fieldName", SqlTypes.BOOLEAN))
+        .keyField(Field.of("diff", "fieldName", SqlTypes.STRING))
+        .build();
+
+    // When:
+    final ConnectSchema result = schema.keySchema();
+
+    // Then:
+    final List<org.apache.kafka.connect.data.Field> fields = result.fields();
+    assertThat(fields, contains(
+        connectField("fieldName", 0, Schema.OPTIONAL_FLOAT64_SCHEMA),
+        connectField("source.fieldName", 1, Schema.OPTIONAL_BOOLEAN_SCHEMA),
+        connectField("diff.fieldName", 2, Schema.OPTIONAL_STRING_SCHEMA)
+    ));
+  }
+
+  @Test
+  public void shouldGetValueSchema() {
```

Comment:
nit: for this test and the one above we should also add some key/value fields respectively to make sure that they do not show up